app/models/miq_queue.rb

```diff
@@ -116,6 +116,11 @@ def self.put(options)
 
     options[:args] = [options[:args]] if options[:args] && !options[:args].kind_of?(Array)
 
+    if !Rails.env.production? && options[:args] &&
+       (arg = options[:args].detect { |a| a.kind_of?(ActiveRecord::Base) && ! a.new_record? })
+      raise "MiqQueue.put(:args =>[]) does not support #{arg.class.name}[#{arg.try(:id)}] objects"
```

Comment:
@kbrock Maybe something like this: 
`raise "MiqQueue.put(:args =>[]) does not support #{arg.class.name} objects, try passing the object(s) class name and id as args to #{options[:class_name]} #{options[:method_name]} instead"`